### PR TITLE
Add terrascan

### PR DESCRIPTION
**Pull request by JonZeolla**
# Contributor Comments
This adds `terrascan` to each `terraform` command

## Pull Request Checklist
Thank you for submitting a contribution to `easy_infra`!

In order to streamline the review of your contribution we ask that you review and comply with the below requirements:
 - [X] Rebase your branch against the latest commit of the target branch, which likely should be `main`
 - [X] If there is an issue associated with your Pull Request, align your PR title with [this documentation](https://help.github.com/en/articles/closing-issues-using-keywords)

### CODE REVIEW

File: tests/terraform/tfsec/insecure.tf [hunk 1]
```diff
@@ -0,0 +1,6 @@
+resource "aws_security_group_rule" "example" {
```
Comment (tp-seiso) [inline tests/terraform/tfsec/insecure.tf:1]:
```suggestion
provider "aws" {
    region = "us-east-1"
}

resource "aws_security_group_rule" "example" {
```

File: tests/terraform/tfsec/insecure.tf [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+    # fails tfscan check AWS006
+    type        = "ingress"
+    cidr_blocks = ["0.0.0.0/0"]
+    description = "This is an example insecure rule for tfsec testing"
```
Comment (tp-seiso) [inline tests/terraform/tfsec/insecure.tf:5]:
```suggestion
    description = "This is an example insecure rule for tfsec testing"
    from_port   = "0"
    to_port     = "0"
    protocol    = "-1"
    security_group_id = "sg-id"
```

File: tests/terraform/tfsec/insecure.tf (hunk 1 above)
Comment (tp-seiso) [inline tests/terraform/tfsec/insecure.tf:1]: This tf file failed validate so it didn't return the tfsec error that is expected. Added the provider and other missing required fields for the resource and it functioned as expected.

```
docker run -v ./tests/terraform/tfsec:/iac seiso/easy_infra /bin/bash -c "terraform --skip-terrascan --skip-checkov init"
INFO:  Passed terraform init initialization
ERROR:  Failed terraform validate validation

Error: Missing required argument

The argument "region" is required, but was not set.
```

File: tasks.py [hunk 3]
```diff
@@ -199,10 +199,38 @@ def run_terraform_tests(*, image: str):
...
+        volumes=volumes,
+        working_dir=working_dir,
+        command=command,
+        environment=environment,
+        expected_exit=1,
+    )
+    num_tests_ran += 1
+
+    # Ensure insecure configurations fail due to terrascan
+    command = "terraform --skip-tfsec --skip-checkov plan -lock=false"
```
Comment (tp-seiso) [inline tasks.py:231]: checkov still runs when the `--skip-checkov` flag is second. This should be switched for this build and addressed later as #64

**Review by tp-seiso — CHANGES_REQUESTED** [review]
A few changes to correct the `tfsec/insecure.tf` file and work around issue #64

File: tasks.py (hunk 3 above)
Comment (JonZeolla) [reply tasks.py:231]: Nice catch; this is a design flaw I'll need to think about.  Probably a while loop processing `"$@"`

File: tests/terraform/tfsec/insecure.tf (hunk 1 above)
Comment (JonZeolla) [reply tests/terraform/tfsec/insecure.tf:5]: Now working as expected; thanks for the catch!

**Comment by JonZeolla** [comment]
@tp-seiso responded to feedback, nice catches

**Review by tpears — APPROVED** [review]
LGTM

**Pull request merged**